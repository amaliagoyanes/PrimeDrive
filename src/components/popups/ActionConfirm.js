import React, {Component} from 'react';
import {
    Dimensions,
    TouchableOpacity,
    Text,
    Button,
    View,
} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import {Row, Col} from 'react-native-easy-grid';
import Modal from 'react-native-root-modal';
import styles from '../../style/styles';
import {scale, scaleText} from '../../scaling'


export default class ActionConfirm extends Component {
    constructor(props) {
        super(props);
        this.state = {
            open: false,
            params: null, text: null
        };

        this.close = this.close.bind(this);
        this.open = this.open.bind(this);
        this.openWithParams = this.openWithParams.bind(this);
    }

    openWithParams(units, uid, data) {
        let params;
        if (data !== undefined)
            params = data;
        else
            params = {amount: units, uid: uid};
        this.setState({
            open: true,
            params: params
        });
    }

    open(text) {
        this.setState({open: true, text: text !== undefined ? text : null})
    }

    close() {
        this.setState({open: false, text: null})
    }

    render() {
        return (
            <Modal
                onRequestClose={this.close}
                visible={this.state.open}
                style={styles.modal}
            >
                <TouchableOpacity style={styles.overlay} onPress={() => {
                    if (!this.props.explicitClosing)
                        this.close();
                }}>
                    <TouchableOpacity activeOpacity={1} onPress={(event) => event.stopPropagation()} style={{
                        position: 'absolute',
                        top: Dimensions.get('screen').height * 0.375,
                        left: Dimensions.get('screen').width * 0.375,
                        width: Dimensions.get('screen').width * 0.25,
                        height: Dimensions.get('screen').height * (this.props.noButtons ? 0.2 : 0.25),
                        backgroundColor: '#e3e5e6'
                    }}>
                        <Row size={0.75} style={{alignItems: 'flex-start', justifyContent: 'flex-end'}}>
                            <TouchableOpacity onPress={this.close}
                                              style={[styles.timesIcon, {
                                                  marginTop: 0, marginLeft: scale(32), padding: scale(5),
                                                  backgroundColor: 'transparent'
                                              }]}>
                                <Icon size={scaleText(20)} name="close" color="#313138"/>
                            </TouchableOpacity>
                        </Row>
                        <Row size={this.props.noButtons ? 2.25 : 1}
                             style={{alignItems: 'center', justifyContent: 'center'}}>
                            {this.state.text !== null ?
                                <Text style={[styles.text, {color: '#313138', fontSize: scaleText(13)}]}>
                                    {this.state.text}</Text> :
                                this.props.children}
                        </Row>
                        {!this.props.noButtons &&
                        <Row size={1.25} style={{alignItems: 'flex-end', justifyContent: 'flex-end'}}>
                            {!this.props.onlyOkButton &&
                            <TouchableOpacity onPress={() => {
                                this.close();
                                if (!!this.props.onCancel) {
                                    this.props.onCancel();
                                }
                            }} style={[styles.tabButton, styles.confirmBtn,
                                {backgroundColor: !!this.props.onlyCancelButton ? '#e84b3a' : 'transparent',}]}>
                                <Text style={[styles.text, {color: !!this.props.onlyCancelButton     ? '#fff': '#313138', fontSize: scaleText(13)}]}>
                                    {this.props.intl !== undefined && this.props.intl.messages["app.cancelUpper"]}</Text>
                            </TouchableOpacity>
                            }
                            {!this.props.onlyCancelButton &&
                            <TouchableOpacity onPress={() => {
                                this.close();
                                this.props.onOk(this.state.params)
                            }}
                                              style={[styles.tabButton, styles.confirmBtn]}>
                                <Text style={[styles.text, {fontSize: scaleText(13)}]}>
                                    {this.props.intl !== undefined && this.props.intl.messages["app.okUpper"]}</Text>
                            </TouchableOpacity>
                            }
                        </Row>
                        }
                    </TouchableOpacity>
                </TouchableOpacity>
            </Modal>
        )
    }
}
