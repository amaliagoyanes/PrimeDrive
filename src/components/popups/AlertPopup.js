import React, {Component} from 'react';
import {
    Dimensions,
    TouchableOpacity,
    Text,
    Button,
    View,
} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import {Row, Col} from 'react-native-easy-grid';
import Modal from 'react-native-root-modal';
import styles from '../../style/styles';
import {scale, scaleText} from '../../scaling'


export default class AlertPopup extends Component {
    constructor(props) {
        super(props);
        this.state = {
            open: false,
            text: null
        };

        this.close = this.close.bind(this);
        this.open = this.open.bind(this);
    }

    open(text) {
        this.setState({open: true, text: text !== undefined ? text : null})
    }

    close() {
        this.setState({open: false, text: null})
    }

    render() {
        return (
            <Modal
                onRequestClose={this.close}
                visible={this.state.open}
                style={styles.modal}
            >
                <TouchableOpacity style={styles.overlay} onPress={() => {
                    if (!this.props.explicitClosing)
                        this.close();
                }}>
                    <TouchableOpacity activeOpacity={1} onPress={(event) => event.stopPropagation()} style={{
                        position: 'absolute',
                        top: Dimensions.get('screen').height * 0.375,
                        left: Dimensions.get('screen').width * 0.25,
                        width: Dimensions.get('screen').width * 0.5,
                        height: Dimensions.get('screen').height * 0.25,
                        backgroundColor: '#e3e5e6'
                    }}>
                        <Row size={0.75} style={{alignItems: 'flex-start', justifyContent: 'flex-end'}}>
                            <TouchableOpacity onPress={this.close}
                                              style={[styles.timesIcon, {
                                                  marginTop: 0, marginLeft: scale(32), padding: scale(5),
                                                  backgroundColor: 'transparent'
                                              }]}>
                                <Icon size={scaleText(20)} name="close" color="#313138"/>
                            </TouchableOpacity>
                        </Row>
                        <Row size={2}
                             style={{alignItems: 'center', justifyContent: 'center'}}>
                                <Text style={[styles.text, {color: '#313138', fontSize: scaleText(13)}]}>
                                    {this.state.text}</Text>
                        </Row>
                        <Row size={1.25} style={{alignItems: 'flex-end', justifyContent: 'flex-end'}}>
                            <TouchableOpacity onPress={this.close} style={[styles.tabButton, styles.confirmBtn, {width: '20%'}]}>
                                <Text style={[styles.text, {fontSize: scaleText(13)}]}>
                                    {this.props.intl !== undefined && this.props.intl.messages["app.okUpper"]}</Text>
                            </TouchableOpacity>
                        </Row>
                    </TouchableOpacity>
                </TouchableOpacity>
            </Modal>
        )
    }
}
