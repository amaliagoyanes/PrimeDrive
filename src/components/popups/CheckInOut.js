import moment from 'moment';
import React, {Component} from 'react';
import {
    AsyncStorage,
    Dimensions,
    TouchableOpacity,
    Text,
    View,
} from 'react-native';
import {FormattedMessage} from 'react-intl';
import {Row, Col} from 'react-native-easy-grid';
import Modal from 'react-native-root-modal';
import styles from '../../style/styles';
import firestack from '../../fbconfig.js'
import {ButtonItem, IconButton, ModalHeader, ModalCloseButton, guid} from '../index';
import {scale, scaleText} from '../../scaling'


export default class CheckInOut extends Component {
    constructor(props) {
        super(props);
        this.state = {
            open: false, name: '', number: '', uid: null, timestamp: null
        };

        this.modalName = 'PUNCH IN OR OUT';

        this.checkInOut = this.checkInOut.bind(this);
        this.close = this.close.bind(this);
        this.open = this.open.bind(this);
    }

    open() {
        this.setState({open: true}, () => this.loadData())
    }

    close() {
        this.setState({open: false})
    }

    loadData() {
        AsyncStorage.getItem('@PrimeDrive:user_info')
            .then(info => {
                info = JSON.parse(info);
                AsyncStorage.getItem('@PrimeDrive:checkin_uid').then(checkin => {
                    this.setState({number: info.number, name: info.name, uid: checkin}, () =>
                        AsyncStorage.setItem('@PrimeDrive:checkin_timestamp').then(timestamp =>
                            this.setState({timestamp: timestamp})).catch()
                    );
                }).catch();
            })
            .catch(err => console.log(err));
    }

    checkInOut() {
        if (this.state.uid === null) { //no checkin
            let uid = guid();
            let checkin_time = moment().unix();
            let data = {
                checkin_time: checkin_time,
                client: this.props.client_id,
                user_id: this.props.user_uid,
                customer_uid: this.props.customer_uid,
                dealer_uid: this.props.dealer_uid,
                status: 'new',
            };
            firestack.database.ref(`/userCheckins/${uid}`).set(data).then(response => {
                AsyncStorage.setItem('@PrimeDrive:checkin_uid', uid.toString()).then(() => {
                    AsyncStorage.setItem('@PrimeDrive:checkin_timestamp', checkin_time.toString()).then(() => {
                        this.setState({uid: uid, timestamp: checkin_time.toString()});
                        this.close();
                    }).catch();
                }).catch();
            }).catch(err => console.log(err));
        } else { //checkout action
            firestack.database.ref(`/userCheckins/${this.state.uid}`).once('value').then((snapshot) => {
                let val = snapshot.val();
                val.checkout_time = moment().unix();
                val.status = 'current';
                firestack.database.ref(`/userCheckins/${this.state.uid}`).set(val).then(response => {
                    AsyncStorage.removeItem('@PrimeDrive:checkin_uid').then().catch();
                    AsyncStorage.removeItem('@PrimeDrive:checkin_timestamp').then().catch();
                    this.setState({uid: null, timestamp: null, open: false});
                })
            });
        }
    }

    render() {
        let active = this.state.uid !== null;
        let diff = this.state.timestamp ? moment().diff(moment.unix(this.state.timestamp), 'minutes') : '';
        return (
            <Modal
                onRequestClose={this.close}
                visible={this.state.open}
                style={styles.modal}
            >
                <TouchableOpacity style={styles.overlay} onPress={this.close}>
                    <TouchableOpacity activeOpacity={1} onPress={(event) => event.stopPropagation()} style={{
                        position: 'absolute',
                        top: Dimensions.get('screen').height * 0.1,
                        left: Dimensions.get('screen').width * 0.1,
                        width: Dimensions.get('screen').width * 0.8,
                        height: Dimensions.get('screen').height * 0.8,
                        backgroundColor: '#e8eaeb',
                        paddingLeft: scale(40),
                        paddingRight: scale(40),
                        paddingTop: scale(0),
                        paddingBottom: scale(50),
                    }}>
                        <ModalCloseButton close={this.close}/>
                        <ModalHeader modalName={this.modalName}/>
                        <Row size={5}/>
                        <Row size={13} style={{alignItems: 'center', justifyContent: 'center'}}>
                            <Col/>
                            <Col size={2}>
                                <Row>
                                    <Col size={1}>
                                        <IconButton style={{
                                            height: scale(90),
                                            marginLeft: 0,
                                            marginRight: 0,
                                            backgroundColor: '#404048'
                                        }}
                                                    text={this.state.name} price={this.state.number} active/>
                                    </Col>
                                    <Col size={3} style={{marginLeft: scale(45), paddingTop: scale(5)}}>
                                        <Row size={1}>
                                            <Text style={[styles.text, {fontSize: scaleText(10), color: '#313138'}]}>KASPER
                                                SJ</Text>
                                        </Row>
                                        <Row size={3}>
                                            <Text
                                                style={[styles.text, {fontSize: scaleText(23), color: '#313138'}]}>
                                                {active ? this.props.intl.messages["app.checkedIn"] :
                                                    this.props.intl.messages["app.checkedOut"]}
                                            </Text>
                                        </Row>
                                    </Col>
                                </Row>
                                <Row>
                                    <ButtonItem text="app.punchIn" onPress={this.checkInOut} disabled={active}
                                                intl={this.props.intl}
                                                style={[styles.modalButton, {
                                                    backgroundColor: !active ? '#3cb671' : '#bec3c7', marginRight: 0
                                                }]}
                                                textStyle={[styles.modalButtonText]}/>
                                </Row>
                                <Row>
                                    <ButtonItem text="app.punchOut" onPress={this.checkInOut} disabled={!active}
                                                intl={this.props.intl}
                                                style={[styles.modalButton, {
                                                    backgroundColor: active ? '#3cb671' : '#bec3c7', marginRight: 0
                                                }]}
                                                textStyle={[styles.modalButtonText]}/>
                                </Row>
                                <Row>
                                    <Text style={[styles.text, {
                                        color: '#313138',
                                        marginTop: scale(40),
                                        fontSize: scaleText(10)
                                    }]}>
                                        {this.state.timestamp !== null && this.props.intl !== undefined &&
                                            this.props.intl.messages["app.currentTimeUpper"]
                                        }
                                        {this.state.timestamp !== null &&
                                        `${parseInt(diff / 60)}:${diff % 60}`
                                        }
                                    </Text>
                                </Row>
                            </Col>
                            <Col/>
                        </Row>
                    </TouchableOpacity>
                </TouchableOpacity>
            </Modal>
        )
    }
}
