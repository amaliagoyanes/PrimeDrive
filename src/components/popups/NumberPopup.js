import React, {Component} from 'react';
import {
    Dimensions,
    TouchableOpacity,
    Text,
    View,
    Alert
} from 'react-native';
import {FormattedMessage} from 'react-intl';
import {Row, Col} from 'react-native-easy-grid';
import styles from '../../style/styles';
import Modal from 'react-native-root-modal';
import {NumericKeyboard} from '../index.js'
import {scale, scaleText} from '../../scaling'


export default class NumberPopup extends Component {
    constructor(props) {
        super(props);
        this.state = {
            input: '', open: false, px: 0, py: 0
        };

        this.numOnPress = this.numOnPress.bind(this);
        this.cleanInput = this.cleanInput.bind(this);
        this.confirm = this.confirm.bind(this);
        this.close = this.close.bind(this);
        this.open = this.open.bind(this);
    }

    open(px, py, uid, negative=false) {
        this.setState({open: true, px: px, py: py, uid: uid, negative: negative})
    }

    close() {
        this.setState({open: false, input: ''})
    }

    numOnPress(text) {
        if (this.props.lockZero && this.state.input.length <= 0 && Number(text) === 0) 
            return;
        let newVal = this.state.input + text,
            max = this.props.max;
        if (max && parseFloat(newVal) > max) {
            this.props.alert(this.props.intl.messages["app.notGreaterThan"] + max);
            return;
        }
        this.setState({input: newVal})
    }

    cleanInput(){
        if (this.props.deleteAll)
            this.setState({input: ''});
        else
            this.setState({input: this.state.input.slice(0, this.state.input.length - 1)});
    }

    confirm(){
        if (this.props.onOk !== undefined)
            this.props.onOk(this.state.negative ? parseFloat(-this.state.input) : parseFloat(this.state.input), null, this.state.uid);
        this.close();
    }

    render() {
        let left = this.props.arrow === 'left' ?
            this.state.px + 10 : this.state.px - Dimensions.get('screen').width * 0.15 - 10;
        let leftArrow = this.props.arrow === 'left' ?
            this.state.px : this.state.px - 15;
        let py = this.state.py > Dimensions.get('screen').height / 2 ? 215 : 50;

        return (
            <Modal
                onRequestClose={this.close}
                visible={this.state.open}
                style={styles.modal}
            >
                <TouchableOpacity style={styles.overlay} onPress={this.close}>
                    <View style={{
                        position: 'absolute',
                        top: this.state.py + 15,
                        left: leftArrow,
                    }}>
                        <View
                            style={this.props.arrow === 'left' ? [styles.triangle, styles.leftTriangle] : styles.triangle}/>
                    </View>
                    <TouchableOpacity activeOpacity={1} onPress={(event) => event.stopPropagation()} style={{
                        position: 'absolute',
                        top: this.state.py - py,
                        left: left,
                        width: Dimensions.get('screen').width * 0.15,
                        height: Dimensions.get('screen').height * 0.4,
                        backgroundColor: '#404048'
                    }}>
                        <NumericKeyboard size={6} numOnPress={this.numOnPress} input={this.state.input}
                                         lockZero={this.props.lockZero}
                                         cleanInput={this.cleanInput} smallBlock intl={this.props.intl}
                                         showDelimiter={this.props.showDelimiter} postfix={this.props.postfix}/>
                        <Row size={1}>
                            <TouchableOpacity style={[styles.tabButton, {width: '100%', backgroundColor: this.state.input.length ? '#2eab61' : '#93D8B1' }]} // '#93D8B1'
                                              onPress={this.confirm} disabled={!this.state.input.length}>
                                <Text style={[styles.text, {fontSize: scaleText(9)}]}>
                                    {this.props.intl !== undefined && this.props.intl.messages["app.okUpper"]}</Text>
                            </TouchableOpacity>
                        </Row>
                    </TouchableOpacity>
                </TouchableOpacity>
            </Modal>
        )
    }
}
