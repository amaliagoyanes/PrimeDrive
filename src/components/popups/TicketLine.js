import React, {Component} from 'react';
import {
    Dimensions,
    TouchableOpacity,
    Text,
    View,
    TextInput,
} from 'react-native';
import {FormattedMessage} from 'react-intl';
import {Row, Col} from 'react-native-easy-grid';
import styles from '../../style/styles';
import Modal from 'react-native-root-modal';
import {NumericKeyboard} from '../index.js';
import {scale, scaleText} from '../../scaling';
import {toFixedLocale} from '../plain/receipts/plainComponents';


export default class TicketLine extends Component {
    constructor(props) {
        super(props);
        this.state = {
            input: '', active: 0, open: false, px: 0, py: 0, uid: null, name: ''
        };

        this.setActive = this.setActive.bind(this);
        this.confirm = this.confirm.bind(this);
        this.close = this.close.bind(this);
        this.open = this.open.bind(this);
    }

    setActive(i) {
        this.setState({active: i})
    }

    open(px, py, amount = 1, name = '', uid = null, note = '', product_price = 0, discount = '', campaign = '') {

        this.setState({open: true, px: px, py: py, name: name, amount: amount, uid: uid, note: note,
                       product_price: product_price, discount: discount, campaign: campaign})
    }

    close() {
        this.setState({open: false})
    }

    confirm() {
        let values = this.controls.getVal();
        if (this.props.onOk !== undefined) {
            this.props.onOk(this.amountCtrl.getVal(), this.state.uid, undefined, values.note, values.discount);
        }
        this.close();
    }

    render() {
        let left = this.props.arrow === 'left' ?
            this.state.px + 10 : this.state.px - Dimensions.get('screen').width / 6 - 10;
        let leftArrow = this.props.arrow === 'left' ?
            this.state.px : this.state.px - 15;
        let isDiscount = this.state.active === 0 && !this.props.readOnly;
        let isCampaign = this.state.campaign && this.state.campaign.amount;
        let campaign_amount = '';
        let campaign_text = '';
        if (isCampaign && this.state.campaign.type === 'discount') {
            campaign_amount = toFixedLocale(this.state.product_price * this.state.amount * (parseFloat(this.state.campaign.amount.replace(',', '.')) / 100));
            campaign_text = `${this.props.intl.messages["app.campaign"]} ${this.state.campaign.name} - ${this.state.campaign.amount}%`;
        }
        if (isCampaign && this.state.campaign.type === 'price') {
            campaign_amount = toFixedLocale(this.state.campaign.amount.replace(',', '.') * this.state.amount);
            campaign_text = `${this.props.intl.messages["app.campaign"]} ${this.state.campaign.name} - ${this.props.intl.messages["app.price"]} ${this.state.campaign.amount} DKK`;
            // style.textDecorationLine = 'line-through';
        }

        return (
            <Modal
                onRequestClose={this.close}
                visible={this.state.open}
                style={styles.modal}
            >
                <TouchableOpacity style={styles.overlay} onPress={this.close}>
                    <View style={{
                        position: 'absolute',
                        top: this.state.py + 5,
                        left: leftArrow
                    }}>
                        <View
                            style={this.props.arrow === 'left' ? [styles.triangle, styles.leftTriangle] : styles.triangle}/>
                    </View>
                    <TouchableOpacity activeOpacity={1} onPress={(event) => event.stopPropagation()} style={{
                        position: 'absolute',
                        top: this.state.py - 20,
                        left: left,
                        width: Dimensions.get('screen').width / 6,
                        height: isDiscount ? 300 : 155,
                        backgroundColor: '#404048'
                    }}>
                        <Row size={!!campaign_amount ? 1 : 2}
                            style={{alignItems: 'center', justifyContent: 'center', padding: scale(5)}}>
                            <Text style={styles.text}>{this.state.name.toUpperCase()}</Text>
                        </Row>
                        {!!campaign_amount &&
                        <Row style={{alignItems: 'center', justifyContent: 'center', marginLeft: scale(5), marginRight: scale(5)}}>
                            <Text style={[styles.text, styles.sectionDiscount]}>{campaign_text}</Text>
                            <Text style={[styles.text, styles.sectionDiscountPrice]}>{this.state.campaign.type==='discount'?'-':''}{campaign_amount} DKK</Text>
                        </Row>
                        }

                        <AmountControl ref={elt => this.amountCtrl = elt} amount={this.state.amount} />
                        <Row style={{backgroundColor: '#6f7076', marginTop: 1}}>
                            <TouchableOpacity
                                style={this.state.active === 0 ? [styles.tabButton, styles.activeColor] : [styles.tabButton]}
                                onPress={() => this.setState({active: 0})}>
                                <Text style={[styles.text, {fontSize: scaleText(9)}]}>
                                    {this.props.intl.messages["app.discountUpper"]}</Text>
                            </TouchableOpacity>
                            <TouchableOpacity
                                style={this.state.active === 1 ? [styles.tabButton, styles.activeColor] : [styles.tabButton]}
                                onPress={() => this.setState({active: 1})}>
                                <Text style={[styles.text, {fontSize: scaleText(9)}]}>
                                    {this.props.intl.messages["app.noteUpper"]}</Text>
                            </TouchableOpacity>
                            <TouchableOpacity style={[styles.tabButton, {backgroundColor: '#2eab61'}]}
                                              onPress={this.confirm}>
                                <Text style={[styles.text, {fontSize: scaleText(9)}]}>
                                    {this.props.intl.messages["app.okUpper"]}
                                </Text>
                            </TouchableOpacity>
                        </Row>
                        <TicketLineControls active={this.state.active} isDiscount={isDiscount} intl={this.props.intl}
                                            discount={this.state.discount} note={this.state.note} alert={this.props.alert}
                                            ref={elt => this.controls = elt} />
                    </TouchableOpacity>
                </TouchableOpacity>
            </Modal>
        )
    }
}

export class TicketLineControls extends Component {
    constructor(props) {
        super(props);
        this.state = {
            input: props.discount || '', text: props.note || ''
        };

        this.numOnPress = this.numOnPress.bind(this);
        this.cleanInput = this.cleanInput.bind(this);
    }

    numOnPress(text) {
        let newVal = this.state.input + text,
            max = 100;
        console.log('this.props.alert ', this.props);
        if (parseFloat(newVal) > max)
            this.props.alert(`${this.props.intl.messages["app.notGreaterThan"]} ${max}`);
        else
            this.setState({input: newVal})
    }

    cleanInput() {
        this.setState({input: ''});
    }

    getVal() {
        return {note: this.state.text, discount: this.state.input};
    }

    render() {
        return (
            <Row size={this.props.isDiscount ? 8 : 2} style={{backgroundColor: '#6f7076', padding: 0}}>
                {this.props.active !== 0 ?
                    <View style={[styles.tabInfo, {alignItems: 'center'}]}>
                        <TextInput value={this.state.text}
                                   style={[styles.text, {fontSize: scaleText(11), width: '100%', height: '100%'}]}
                                   multiline={true}
                                   onChangeText={(text) => this.setState({text})}/>
                    </View> :
                    this.props.readOnly ?
                        <View style={[styles.tabInfo, {alignItems: 'flex-end', padding: 10}]}>
                            <Text style={[styles.text, {fontSize: scaleText(13)}]}>25%</Text>
                        </View> :
                        <Col>
                            <Row size={5}>
                                <NumericKeyboard size={6} numOnPress={this.numOnPress} intl={this.props.intl}
                                                 input={this.state.input} inputBackgroundColor='#6f7076'
                                                 cleanInput={this.cleanInput} smallBlock
                                                 showDelimiter={true} postfix={'%'}/>
                            </Row>
                        </Col>
                }
            </Row>
        )
    }
}

export class AmountControl extends Component {
    constructor(props) {
        super(props);
        this.state = {
            amount: props.amount || 1,
            maxAmount: props.maxAmount

        };

        this.getVal = this.getVal.bind(this);
        this.onPlusPress = this.onPlusPress.bind(this);
    }

    getVal() {
        return this.state.amount;
    }

    onPlusPress() {
        if (this.state.maxAmount) {
            if (this.state.maxAmount > this.state.amount)
                this.setState({amount: this.state.amount + 1});
        }
        else
            this.setState({amount: this.state.amount + 1})
    }

    render() {
        return (
            <Row style={{alignItems: 'center'}}>
                <TouchableOpacity name="minus" style={[styles.tabButton]}
                                  onPress={() => this.state.amount !== 1 ? this.setState({amount: this.state.amount - 1}) : ''}>
                    <Text style={[styles.text]}>-</Text>
                </TouchableOpacity>
                <TouchableOpacity style={[styles.tabButton]}>
                    <Text style={styles.text}>{this.state.amount}</Text>
                </TouchableOpacity>
                <TouchableOpacity name="plus" style={[styles.tabButton]}
                                  onPress={this.onPlusPress}>
                    <Text style={[styles.text]}>+</Text>
                </TouchableOpacity>
            </Row>
        )
    }
}