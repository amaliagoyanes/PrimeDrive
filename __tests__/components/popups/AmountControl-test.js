'use strict';

import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import {AmountControl} from '../../../src/components/popups/TicketLine';


it('renders correctly', () => {
    const tree = shallow(
        <AmountControl />);
    expect(tree).toMatchSnapshot();
    expect(tree.find('TouchableOpacity').length).toEqual(3);
});

it('renders correctly with amount', () => {
    const tree = shallow(
        <AmountControl amount={2}/>);
    expect(tree).toMatchSnapshot();
});


it('value cannot be < 0', () => {
    const tree = shallow(
        <AmountControl/>);
    tree.find('[name="minus"]').simulate('press');
    expect(tree).toMatchSnapshot();
    expect(tree.instance().getVal()).toEqual(1);
});


it('value cannot be > maxAmount', () => {
    const tree = shallow(
        <AmountControl maxAmount={2}/>);
    tree.find('[name="plus"]').simulate('press');
    tree.find('[name="plus"]').simulate('press');
    expect(tree).toMatchSnapshot();
    expect(tree.instance().getVal()).toEqual(2);
});


