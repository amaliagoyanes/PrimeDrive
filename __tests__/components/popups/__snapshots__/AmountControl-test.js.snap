// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RowNB
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    name="minus"
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#404048",
          "justifyContent": "center",
          "width": "34%",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
        ]
      }
    >
      -
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#404048",
          "justifyContent": "center",
          "width": "34%",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Gotham-Book",
          "fontSize": 14,
        }
      }
    >
      1
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    name="plus"
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#404048",
          "justifyContent": "center",
          "width": "34%",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
        ]
      }
    >
      +
    </Text>
  </TouchableOpacity>
</RowNB>
`;

exports[`renders correctly with amount 1`] = `
<RowNB
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    name="minus"
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#404048",
          "justifyContent": "center",
          "width": "34%",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
        ]
      }
    >
      -
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#404048",
          "justifyContent": "center",
          "width": "34%",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Gotham-Book",
          "fontSize": 14,
        }
      }
    >
      2
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    name="plus"
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#404048",
          "justifyContent": "center",
          "width": "34%",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
        ]
      }
    >
      +
    </Text>
  </TouchableOpacity>
</RowNB>
`;

exports[`value cannot be < 0 1`] = `
<RowNB
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    name="minus"
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#404048",
          "justifyContent": "center",
          "width": "34%",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
        ]
      }
    >
      -
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#404048",
          "justifyContent": "center",
          "width": "34%",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Gotham-Book",
          "fontSize": 14,
        }
      }
    >
      1
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    name="plus"
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#404048",
          "justifyContent": "center",
          "width": "34%",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
        ]
      }
    >
      +
    </Text>
  </TouchableOpacity>
</RowNB>
`;

exports[`value cannot be > maxAmount 1`] = `
<RowNB
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    name="minus"
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#404048",
          "justifyContent": "center",
          "width": "34%",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
        ]
      }
    >
      -
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#404048",
          "justifyContent": "center",
          "width": "34%",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Gotham-Book",
          "fontSize": 14,
        }
      }
    >
      2
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    name="plus"
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#404048",
          "justifyContent": "center",
          "width": "34%",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
        ]
      }
    >
      +
    </Text>
  </TouchableOpacity>
</RowNB>
`;
